config/config.go

```diff
@@ -111,6 +117,140 @@ func GetNodeConfig(nodeConf *NodeConfig, cmConf *cmcfg.Config) {
 	}
 }
 
+// UpdateNodeConfigWithFlags updates NodeConfig with values from command line flags.
+func UpdateNodeConfigWithFlags(nodeConf *NodeConfig, flags *pflag.FlagSet) (err error) {
+	updateBoolFlag := func(flagName string, updateFunc func(bool)) {
+		if flag := flags.Lookup(flagName); flag.Changed {
+			updateFunc(flag.Value.String() == "true")
+		}
+	}
+
+	updateStringFlag := func(flagName string, updateFunc func(string)) {
+		if flag := flags.Lookup(flagName); flag.Changed {
+			updateFunc(flag.Value.String())
+		}
+	}
+
+	updateDurationFlag := func(flagName string, updateFunc func(time.Duration) error) error {
```

Comment:
All the lambdas provided as `updateFunc` unconditionally return `nil`, so we might as well remove this param.
```suggestion
	updateDurationFlag := func(flagName string, updateFunc func(time.Duration)) error {
```